vms/avm/service.go

```diff
@@ -92,16 +95,16 @@ func (s *Service) GetBlock(_ *http.Request, args *api.GetBlockArgs, reply *api.G
 				return err
 			}
 		}
-		reply.Block = block
-		return nil
-	}
-
-	reply.Block, err = formatting.Encode(args.Encoding, block.Bytes())
-	if err != nil {
-		return fmt.Errorf("couldn't encode block %s as string: %w", args.BlockID, err)
+		result = block
+	} else {
+		result, err = formatting.Encode(args.Encoding, block.Bytes())
+		if err != nil {
+			return fmt.Errorf("couldn't encode block %s as string: %w", args.BlockID, err)
+		}
```

Comment:
nit: we could move this error and the one above (line 94-96) outside of the if else.

Reply:
The above error is in a for loop - so I don't think it makes sense to refactor this instance